packages/block-library/src/search/index.php

```diff
@@ -283,3 +306,36 @@ function get_border_color_classes_for_block_core_search( $attributes ) {
 	}
 	return $border_color_classes;
 }
+
+/**
+ * Returns color classnames depending on whether there are named or custom text and background colors.
+ *
+ * @param array $attributes The block attributes.
+ *
+ * @return string The color classnames to be applied to the block elements.
+ */
+function get_color_classes_for_block_core_search( $attributes ) {
```

Comment:
If we are enabling gradient backgrounds for buttons we'll need to update this function to generate the appropriate gradient CSS class. It should effectively match the `useColorProps` hook from the editor.